src/server/enterprise/testing/cmd_test.go

```diff
@@ -234,3 +235,113 @@ func TestLoginPachd(t *testing.T) {
 		"user", tu.DexMockConnectorEmail,
 	).Run())
 }
+
+// Tests synching contexts from the enterprise server
+func TestSyncContexts(t *testing.T) {
+	resetClusterState(t)
+	defer resetClusterState(t)
+
+	id := tu.UniqueString("cluster")
+	clusterId := tu.UniqueString("clusterDeploymentId")
+
+	// register a new cluster
+	require.NoError(t, tu.BashCmd(`
+		echo {{.license}} | pachctl enterprise activate
+		echo {{.token}} | pachctl auth activate --enterprise --issuer http://pach-enterprise.enterprise:658 --supply-root-token
+		pachctl enterprise register --id {{.id}} --enterprise-server-address grpc://pach-enterprise.enterprise:650 --pachd-address grpc://pachd.default:650 --pachd-user-address grpc://pachd.default:655 --cluster-deployment-id {{.clusterId}} 
+		`,
+		"id", id,
+		"token", tu.RootToken,
+		"license", tu.GetTestEnterpriseCode(t),
+		"clusterId", clusterId,
+	).Run())
+
+	// assert the registered cluster isn't reflected in the user's config
+	require.YesError(t, tu.BashCmd(`
+		pachctl config list context | match {{.id}}
+		`,
+		"id", id,
+	).Run())
+
+	// sync contexts and assert that the newly registered cluster is accessible
+	require.NoError(t, tu.BashCmd(`
+		pachctl enterprise sync-contexts
+		pachctl config list context | match {{.id}}
+		pachctl config get context {{.id}} | match "\"pachd_address\": \"grpc://pachd.default:655\"" 
+		pachctl config get context {{.id}} | match "\"cluster_deployment_id\": \"{{.clusterId}}\""
+		pachctl config get context {{.id}} | match "\"source\": \"IMPORTED\","
+		`,
+		"id", id,
+		"clusterId", clusterId,
+	).Run())
+
+	// re-register cluster with the same cluster ID and new user address
+	// the user-address should be updated on sync
+	require.NoError(t, tu.BashCmd(`
+		pachctl license update-cluster --id {{.id}} --user-address {{.userAddress}}
+		pachctl enterprise sync-contexts
+		pachctl config get context {{.id}} | match "\"pachd_address\": \"{{.userAddress}}\"" 
+		`,
+		"id", id,
+		"token", tu.RootToken,
+		"license", tu.GetTestEnterpriseCode(t),
+		"clusterId", clusterId,
+		"userAddress", "grpc://pachd.default:700",
+	).Run())
+
+	// re-register cluster with a new cluster ID
+	// the cluster id should be updated and the session token should be set to empty
+	// TODO(acohen4): set session_token so that it can be unset
```

Comment:
@actgardner didn't find a good way to do this. I think to test this we would need both an Active Cluster that we can log into, and be able to redeploy that cluster so that it has a new deployment ID. 

What do you think?

Reply:
Yeah I don't know that we have a good easy way to test that edge case, I'm OK to punt on it for now.